covalent_indexer_test.go

```diff
@@ -50,110 +51,350 @@ func TestCovalentIndexer_SetWSSender_SetTwoConsecutiveWebSockets_ExpectFirstOneC
 	ci, _ := covalent.NewCovalentDataIndexer(
 		&mock.DataHandlerStub{},
 		&http.Server{
-			Addr: "localhost:8080",
-		})
-	called := atomic.Flag{}
-	called.Unset()
-	wss := &ws.WSSender{
-		Conn: &mock.WSConnStub{
-			CloseCalled: func() error {
-				called.Set()
-				return nil
-			},
+			Addr: "localhost:21119",
+		},
+	)
+	defer ci.Close()
+
+	called1 := atomic.Flag{}
+	called1.Unset()
+
+	called2 := atomic.Flag{}
+	called2.Unset()
+
+	wss1 := &mock.WSConnStub{
+		CloseCalled: func() error {
+			called1.Set()
+			return nil
 		},
 	}
 
-	ci.SetWSSender(nil)
-	require.False(t, called.IsSet())
-	ci.SetWSSender(wss)
-	require.False(t, called.IsSet())
-	ci.SetWSSender(wss)
-	require.True(t, called.IsSet())
+	wss2 := &mock.WSConnStub{
+		CloseCalled: func() error {
+			called2.Set()
+			return nil
+		},
+	}
+
+	go ci.SetWSSender(nil)
+	time.Sleep(time.Millisecond * covalent.RetrialTimeoutMS)
+	require.False(t, called1.IsSet())
+	require.False(t, called2.IsSet())
+
+	go ci.SetWSSender(wss1)
+	time.Sleep(time.Millisecond * covalent.RetrialTimeoutMS)
+	require.False(t, called1.IsSet())
+	require.False(t, called2.IsSet())
+
+	go ci.SetWSSender(wss2)
+	time.Sleep(time.Millisecond * covalent.RetrialTimeoutMS)
+	require.True(t, called1.IsSet())
+	require.False(t, called2.IsSet())
 }
 
-func TestCovalentIndexer_sendBlockResultToCovalent_InvalidBlock_ExpectNoSentMessage(t *testing.T) {
+func TestCovalentIndexer_SetWSReceiver_SetTwoConsecutiveWebSockets_ExpectFirstOneClosed(t *testing.T) {
 	ci, _ := covalent.NewCovalentDataIndexer(
 		&mock.DataHandlerStub{},
 		&http.Server{
-			Addr: "localhost:8080",
-		})
-	called := atomic.Flag{}
-	called.Unset()
-	wss := &ws.WSSender{
-		Conn: &mock.WSConnStub{
-			WriteMessageCalled: func(messageType int, data []byte) error {
-				called.Set()
-				return nil
-			},
+			Addr: "localhost:21119",
+		},
+	)
+	defer ci.Close()
+
+	called1 := atomic.Flag{}
+	called1.Unset()
```

Comment:
you can remove the unset after init, as it is already unset.

Reply:
Removed unset here and everywhere else in tests